storage/innobase/fil/fil0fil.cc

```diff
@@ -557,8 +557,8 @@ static bool fil_node_open_file(fil_node_t* node)
 			&success);
 	}
 
-	ut_a(success);
-	ut_a(node->is_open());
+	if (!success || !node->is_open())
+		return false;
```

Comment:
Can we have `!success && node->is_open()`? I hope not. It would be better to retain an assertion for that, or to explicitly `goto fail` here.